airbyte-webapp/src/packages/cloud/services/auth/AuthService.tsx

```diff
@@ -147,26 +147,34 @@ export const AuthenticationProvider: React.FC<React.PropsWithChildren<unknown>>
   const stateRef = useRef(state);
   stateRef.current = state;
 
-  useEffectOnce(() => {
-    return auth.onAuthStateChanged(async (currentUser) => {
-      // We want to run this effect only once on initial page opening
+  const onAuthStateChange = useCallback(
+    async (currentUser) => {
       if (!stateRef.current.inited) {
         if (stateRef.current.currentUser === null && currentUser) {
           await onAfterAuth(currentUser);
         } else {
           authInited();
         }
       }
+    },
+    [onAfterAuth, authInited]
+  );
+
+  useEffectOnce(() => {
+    return auth.onAuthStateChanged(async (currentUser) => {
+      // We want to run this effect only once on initial page opening
+      onAuthStateChange(currentUser);
     });
   });
 
+  // Check auth state on window focus, in case the user logged out in a different tab
   useEffect(() => {
     const onFocus = async () => {
       return auth.onAuthStateChanged(async (currentUser) => {
         if (!currentUser) {
           loggedOut();
         } else {
-          await onAfterAuth(currentUser);
+          onAuthStateChange(currentUser);
```

Comment:
with this new PR changes, async is redundant at `const onFocus = async` and `return auth.onAuthStateChanged(async`

Reply:
Thanks for catching that, indeed there are several unnecessary `async` keywords, will clean those up 🙂

Reply:
no worries, i just came across the repo today and was checking some pull requests.